starter-code/database.js

```diff
@@ -1,145 +1,173 @@
-        
-        callback('Error listing users');
+        database.collection(users).find(callback)
+
+        // callback('Error listing users');
       }
-    });
+    })
   }
 
-  deleteUser( firstName, callback = (error, result) => {}) {
+  deleteUser (firstName, callback = (error, result) => { if (error) callback(error) }) {
     this.connect((error, database) => {
-      if (error){
-        callback(error);
+      if (error) {
+        callback(error)
       } else {
         //  LAB 3
         // Implement the query to delete a user
         // firstName is the name of user that we want to delete
         // remeber once it's finish to comment callback('Error deleting user');
-        
-        callback('Error deleting user');
+
+        database.collection(users).deleteOne({'firstName': firstName}, callback)
+
+        // callback('Error deleting user');
       }
-    });
+    })
   }
 
-  insertProduct(product, callback = (error, result) => {}){
+  insertProduct (product, callback = (error, result) => { if (error) callback(error) }) {
     this.connect((error, database) => {
-      if (error){
-        callback(error);
+      if (error) {
+        callback(error)
       } else {
         // LAB 4
         // Implement the query to insert a product
         // product is the document to insert
         // remeber once it's finish to comment callback('Error inserting product');
-        
-        callback('Error inserting product');
+
+        database.collection(products).insertOne(product, callback)
+
+        // callback('Error inserting product');
       }
-    });
+    })
   }
 
-  listProducts(callback = (error, products) => {}) {
+  listProducts (callback = (error, products) => { if (error) callback(error) }) {
     this.connect((error, database) => {
-      if (error){
-        callback(error);
+      if (error) {
+        callback(error)
       } else {
         // LAB 5
         // Implement the query to list all products
         // remeber once it's finish to comment callback('Error listing products');
-        
-        callback('Error listing products');
+
+        database.collection(products).find(callback)
+
+        // callback('Error listing products');
       }
-    });
+    })
   }
 
-  deleteProduct( productName, callback = (error, result) => {}) {
+  deleteProduct (productName, callback = (error, result) => { if (error) callback(error) }) {
     this.connect((error, database) => {
-      if (error){
-        callback(error);
+      if (error) {
+        callback(error)
       } else {
         // LAB 6
         // Implement the query to delete a product
-        // productName is the name of the producto to delete 
+        // productName is the name of the producto to delete
         // remeber once it's finish to comment callback('Error deleting product');
-        
-        callback('Error deleting product');
+
+        console.log(productName)
```

Comment:
Clean the code before commiting, do not push code with console outputs like this. :)